cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/AnomalyDetectorState.java

```diff
@@ -87,6 +104,9 @@ void onAnomalyHandle(Anomaly anomaly, AnomalyState.Status status) {
     AnomalyState recentAnomalyState = _recentAnomaliesByType.get(anomalyType).get(anomalyId);
     if (recentAnomalyState != null) {
       recentAnomalyState.setStatus(status);
+      if (status == AnomalyState.Status.FIX_STARTED) {
+        _ongoingSelfHealingAnomaly.set(_recentAnomaliesByType.get(anomalyType).remove(anomalyId));
```

Comment:
What if `_recentAnomaliesByType.get(anomalyType).remove(anomalyId)` returns `null` due to updating `_recentAnomaliesByType` after https://github.com/linkedin/cruise-control/pull/756/files#diff-a05de16a44ab4fe3089e7064a288b471R104 via `addAnomalyDetection`?

Moreover, after the anomaly is added, but before it is handled it can be purged from `_recentAnomaliesByType`. Looks like we won't have information about such anomalies. Is it possible to ensure that we always have the ongoing anomaly information?